src/components/SingleArticle.jsx

```diff
@@ -6,32 +6,72 @@ import * as api from '../api'
 export default function SingleArticle () {
     const [article, selectArticle] = useState([])
     const [isLoading, setIsLoading] = useState(true)
-
+    const [addVotes, setAddVotes] = useState(false)
+    const [subVotes, setSubVotes] = useState(false)
     const { article_id } = useParams()
 
     useEffect(() => {
         setIsLoading(true)
         api.fetchSingleArticle(article_id).then(({data}) => {
-            selectArticle(data.article.article[0])
+            selectArticle(data.article.article)
             setIsLoading(false)
         })
     }, [article_id]
     )
 
+    const upVote = (article_id) => {
+        if(addVotes === false) {
+            setAddVotes(true)
+                selectArticle((article) => {
+                    return article.map((votes) => { 
+                        console.log(article)
+                        return {...votes, votes: article[0].votes + 1}
+                    })
+            })  
+            api.incVote(article_id)    
+        } else {
+            setAddVotes(false)
+            downVote(article[0].article_id)
+        }
+    }
+  
+        const downVote = (article_id) => {
+            if(subVotes === false) {
+                setSubVotes(true)
+                    selectArticle((article) => {
+                        return article.map((votes) => { 
+                            console.log(article)
+                            return {...votes, votes: article[0].votes - 1}
+                        })
+                    })
+                    api.decVote(article_id)  
```

Comment:
Needs some error handling for when the api request fails - at the moment, if looks like its gone well until you refresh the page